app/models/group_user.rb

```diff
@@ -48,12 +50,13 @@ def self.update_first_unread_pm(last_seen, limit: 10_000)
           INNER JOIN topics t ON t.id = tag.topic_id
           INNER JOIN users u ON u.id = gu2.user_id
           LEFT JOIN topic_users tu ON t.id = tu.topic_id AND tu.user_id = gu2.user_id
+          LEFT JOIN group_users gu3 ON gu3.group_id IN (:whisperers_group_ids) AND gu3.user_id = gu2.user_id
```

Comment:
I'm wondering why we need to join against the `group_users` table when we're already selecting from the `group_users` table and can rely on a filter instead.